src/Models/Feature.php

```diff
@@ -4,25 +4,43 @@
 
 namespace JustSteveKing\Laravel\FeatureFlags\Models;
 
+use Carbon\Carbon;
+use Exception;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Support\Facades\App;
 use JustSteveKing\Laravel\FeatureFlags\Models\Concerns\NormaliseName;
 use JustSteveKing\Laravel\FeatureFlags\Models\Builders\FeatureBuilder;
 
 class Feature extends Model
 {
     use NormaliseName;
-    
+
     protected $fillable = [
         'name',
         'description',
         'active',
+        'expires_at'
     ];
 
     protected $casts = [
         'active' => 'boolean',
+        'expires_at' => 'datetime'
     ];
 
+    public static function booted()
+    {
+        static::retrieved(function(Feature $feature) {
+            if(
+                config('feature-flags.enable_time_bombs')
+                && ! App::environment(config('feature-flags.time_bomb_environments'))
+            ) {
+                return (! is_null($feature->expires_at) && Carbon::now() >= $feature->expires_at) ? throw new Exception(sprintf('The Feature has expired - %s', $feature->name)) : true;
```

Comment:
This ternary expression is a little strange since the `if` branch essentially evaluates to

```php
return throw new Exception(...);
```

I think expanding this out to an actual if-statement would make this much more readable. The condition can also be further simplified by using `Carbon::isAfter(...)`. This would allow us to get rid of the null check since `Carbon::isAfter(null)` evaluates to `false`.

```php
$featureHasExpired = Carbon::now()->isAfter($feature->expired_at);

if ($featureHasExpired) {
    throw new Exception(sprintf('The Feature has expired - %s', $feature->name));
}

return true;
```

Reply:
Cool, I've updated this :) I think it's because I'm used to using ternary ifs in JS more often :)